File: longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let prefix = '';
    if (strs.length === 0){
        return '';
    }
    for (let i = 0; i < strs[0].length; i++){
        let letter = '';
        for (let string of strs){
            if (letter === ''){
                letter  = string[i];
            } else if (string[i] !== letter){
                return prefix;
            }
        }
        prefix += letter;
    }
    return prefix;
};

File: longestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ''
        if len(strs) == 1:
            return strs[0]
        prefix = ''
        index = 0
        loop = True
        while loop:
            curr = ''
            for i in strs:
                if i == '' or index >= len(i):
                    loop = False
                    return prefix
                if curr == '':
                    #print('checking 0')
                    curr = i[index]
                else:
                    if i[index] != curr:
                        return prefix
                    if index == len(i) - 1:
                        loop = False
            # print curr, prefix, i
            prefix += curr
            index +=1
        
        return prefix 
